compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -1596,16 +1596,16 @@ private void validateRemoteFunctionAttachedToObject(BLangFunction funcNode, BObj
     }
 
     private void validateResourceFunctionAttachedToObject(BLangFunction funcNode, BObjectTypeSymbol objectSymbol) {
+        if (Symbols.isFlagOn(objectSymbol.flags, Flags.SERVICE)
```

Comment:
This doesn't seem like the right place for this check, given that the name is `validateResourceFunctionAttachedToObject`.

Shall we move this to `validateFunctionsAttachedToObject` or introduce a new method `validateServiceMethodQualifiers` and call it there?